gmso/utils/topology_writer.py

```diff
@@ -0,0 +1,134 @@
+BOND_TYPE_ATTRIBS = ['name', 'type1', 'type2']
+ANGLE_TYPE_ATTRIBS = ['name', 'type1', 'type2', 'type3']
+DIHEDRAL_TYPE_ATTRIBS = ['name', 'type1', 'type2', 'type3', 'type4']
+
+
+def _parse_parameters_units(collection, check_consistency=False):
+    for expression in collection:
+        expression_symbols = sympify(expression).free_symbols
+        param_units = {}
+        for a_type in collection[expression]:
+            parameters = expression_symbols - a_type.independent_variables
+            for param in parameters:
+                param_units[str(param)] = str(a_type.parameters[str(param)].units)
+            if not check_consistency:
+                break
+    return param_units
+
+
+def _populate_attribs(etree_el, obj, attribs_list):
+    if hasattr(obj, 'member_types'):
+        for i, member in enumerate(getattr(obj, 'member_types')):
+            setattr(obj, 'type{}'.format(i+1), obj.member_types[i])
+
+    for _attrib in attribs_list:
+        val = getattr(obj, _attrib)
+        if type(val) == u.unyt_quantity:
+            val = str(val.value)
+        etree_el.attrib[_attrib] = str(val)
+
+    if len(obj.parameters) > 0:
+        parameters = etree.SubElement(etree_el, 'Parameters')
+        for name, val in obj.parameters.items():
+            parameter = etree.SubElement(parameters, 'Parameter')
+            parameter.attrib['name'] = name
+            if type(val) == u.unyt_quantity:
+                parameter.attrib['value'] = str(val.value)
+            else:
+                parameter.attrib['value'] = str(val)
+    if hasattr(obj, 'member_types'):
+        for i, member in enumerate(getattr(obj, 'member_types')):
+            delattr(obj, 'type{}'.format(i+1))
+
+
+def _assign_expressions(parent_tag, tag_name, expression):
+    this_tag = etree.SubElement(parent_tag, tag_name)
+    this_tag.attrib['expression'] = expression
+    return this_tag
+
+
+def _assign_parameters(params_unit_dict, parent_tag):
+    for param, unit in params_unit_dict.items():
+        param_unit_el = etree.SubElement(parent_tag, 'ParametersUnitDef')
+        param_unit_el.attrib['parameter'] = param
+        param_unit_el.attrib['unit'] = unit
+
+
+def _create_element_tree(types_dict, parent_el, types):
+    _type = types[0:-1]
+    attributes = {'AtomType': ATOM_TYPE_ATTRIBS,
+                  'BondType': BOND_TYPE_ATTRIBS,
+                  'AngleType': ANGLE_TYPE_ATTRIBS,
+                  'DihedralType': DIHEDRAL_TYPE_ATTRIBS}
+    for expr in types_dict:
+        # Create AtomTypes
+        types_grp = _assign_expressions(parent_el, types, expr)
+        params_unit_dict = _parse_parameters_units(types_dict)
+        _assign_parameters(params_unit_dict, types_grp)
+        for core_type in types_dict[expr]:
+            type_el = etree.SubElement(types_grp, _type)
+            _populate_attribs(type_el, core_type, attributes[_type])
+
+
+class TopologyXMLWriter:
+    @staticmethod
+    def write(**kwargs):
```

Comment:
I generally caution against functions that take only `**kwargs` unless directly forwarding those to another function. It's essentially impossible to document a method's behavior when it accepts only `**kwargs`. There are several fields like "name" and "version" that take on default values. Maybe consider making those explicit keyword arguments instead of `**kwargs`.

Another option would be to accept an instance of some data class that supports specific keys (e.g. "name" and "version" but not "my_random_key") if you have a codified schema.

Also, if a user provides an unsupported keyword argument, I didn't see any kind of validation or warning/error for that.